### PR TITLE
Viite 1529 ui bugs after major refactoring

### PR DESCRIPTION
**Pull request by pedrosag**
UI tests are passing:

image

And backend tests are passing too ( http://172.17.201.56:8888/job/Run%20OTH%20+%20Viite%20Tests/1448/console )

### CODE REVIEW

File: viite-UI/src/model/SelectedLinkProperty.js [hunk 1]
```diff
@@ -178,6 +180,18 @@
         }
     };
 
+    var findUnknown = function(linkId, id) {
+      var permanent = !_.isUndefined(linkId) ? _.uniq(roadCollection.getTmpByLinkId([linkId]), _.isEqual) : (!_.isUndefined(id) ? _.uniq(roadCollection.getTmpById([id]), _.isEqual) : [] );
+      var temporary = !_.isUndefined(linkId) ? _.uniq(roadCollection.getTmpByLinkId([linkId]), _.isEqual) : (!_.isUndefined(id) ? _.uniq(roadCollection.getTmpById([id]), _.isEqual) : [] );
```
Comment (leirasf) [inline viite-UI/src/model/SelectedLinkProperty.js]: This two lines seems to be duplicated.

File: viite-UI/src/view/LinkPropertyForm.js [hunk 2]
```diff
@@ -686,14 +686,17 @@
...
-          eventbus.once('linkProperties:unknownsTreated', function () {
-            rootElement.find('.link-properties button.continue').attr('disabled', true);
-            eventbus.trigger('linkProperties:deselectFeaturesSelected');
-            applicationModel.setSelectionType(selectionType.Unknown);
-            applicationModel.setContinueButton(false);
-            eventbus.trigger('linkProperties:highlightSelectedFloatingFeatures');
-            eventbus.trigger('linkProperties:activateInteractions');
-            eventbus.trigger('linkProperties:deactivateDoubleClick');
+          eventbus.once('linkProperties:unknownsTreated', function (unknowns) {
+            //GUI - this seems to fix the problem of the unknowns being drawn behing the floatings
```
Comment (leirasf) [inline viite-UI/src/view/LinkPropertyForm.js]: Why do you have your name on the comment?

File: viite-UI/src/view/LinkPropertyLayer.js [hunk 3]
```diff
@@ -256,11 +275,21 @@
...
       //Multi is the one en charge of defining if we select just the feature we clicked or all the overlapping
-      //multi: true,
+      multi: true,
       //This will limit the interaction to the specific layer, in this case the layer where the roadAddressLinks are drawn
-      layer: [roadLayer.layer, floatingMarkerLayer, anomalousMarkerLayer, greenRoadLayer, pickRoadsLayer, geometryChangedLayer, suravageRoadLayer, historicRoadsLayer],
+      layers: [roadLayer.layer, floatingMarkerLayer, anomalousMarkerLayer, greenRoadLayer, pickRoadsLayer, geometryChangedLayer, suravageRoadLayer, historicRoadsLayer],
       //Limit this interaction to the singleClick
       condition: ol.events.condition.singleClick,
+      filter: function(feature, layer) {
+        if(applicationModel.getSelectionType().value === selectionType.Floating.value){
```
Comment (leirasf) [inline viite-UI/src/view/LinkPropertyLayer.js]: You can save `applicationModel.getSelectionType().value` on a variable.

File: viite-UI/src/model/SelectedLinkProperty.js [hunk 4]
```diff
@@ -178,6 +180,18 @@
...
+      var permUnknown = _.reject(permanent, function(road) {
+        return road.roadLinkType !== RoadLinkType.UnknownRoadLinkType.value && road.anomaly !== Anomaly.None.value;
```
Comment (Campanha) [inline viite-UI/src/model/SelectedLinkProperty.js]: if temporarary and permanent are equal then the "return road.getData().roadLinkType" line 187 and "return road.getData().roadLinkType" line 190 will lead to errors

**Pull request merged**